### ISSUE TITLE
[NNPack] Runtime error with padded `Conv1d` and `>=16` batch size

### ISSUE THREAD

**Issue by ganler** [pytorch/pytorch#90142]
### 🐛 Describe the bug

When using PyTorch without MKL (compiled from scratch with `USE_MKLDNN=0`), I cannot run conv1d when:
1. padding size is non-zero;
2. and batch size >= 16

This is very similar to a previous bug https://github.com/pytorch/pytorch/issues/49462 (fixed by @malfet ) so I guess it is a dispatching problem.

```python
import torch

# N C L
print("✔️ No padding")
print(f"{torch.nn.Conv1d(1, 1, 1, padding=0)(torch.ones(16, 1, 1)).shape = }")
print("✔️ Padding but let the batch size < 16 is okay.")
print(f"{torch.nn.Conv1d(1, 1, 1, padding=1)(torch.ones(15, 1, 1)).shape = }")
print("❌ Padding and let the batch size > 16 is **NOT** okay.")
print(f"{torch.nn.Conv1d(1, 1, 1, padding=1)(torch.ones(16, 1, 1)).shape = }")
```

### Versions

```
PyTorch version: 1.14.0a0+gitf7520cb
Is debug build: False
CUDA used to build PyTorch: Could not collect
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.5 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Clang version: 10.0.0-4ubuntu1 
CMake version: version 3.22.1
Libc version: glibc-2.31

Python version: 3.8.15 (default, Nov 24 2022, 15:19:38)  [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-5.4.0-131-generic-x86_64-with-glibc2.17
Is CUDA available: False
CUDA runtime version: 11.6.112
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: 
GPU 0: NVIDIA RTX A6000
GPU 1: NVIDIA RTX A6000
GPU 2: NVIDIA RTX A6000
GPU 3: NVIDIA RTX A6000

Nvidia driver version: 510.68.02
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.2.2
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.2.2
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.2.2
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.2.2
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.2.2
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.2.2
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.2.2
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: False

Versions of relevant libraries:
[pip3] numpy==1.23.4
[pip3] torch==1.14.0a0+gitf7520cb
[conda] blas                      1.0                         mkl  
[conda] magma-cuda116             2.6.1                         1    pytorch
[conda] mkl                       2021.4.0           h06a4308_640  
[conda] mkl-include               2022.1.0           h06a4308_224  
[conda] mkl-service               2.4.0            py38h7f8727e_0  
[conda] mkl_fft                   1.3.1            py38hd3c417c_0  
[conda] mkl_random                1.2.2            py38h51133e4_0  
[conda] numpy                     1.23.4           py38h14f4228_0  
[conda] numpy-base                1.23.4           py38h31eccc5_0  
[conda] torch                     1.14.0a0+gitf7520cb           dev_0    <develop>
```

cc @VitalyFedyunin @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

**Comment by jingxu10** [comment]
@CaoE

**Comment by jbschlosser** [comment]
Hey @ganler, thanks for the investigation and report! I agree it does appear to be a dispatching issue.

**Comment by CaoE** [comment]
@ganler Is there any error info. with padding and batch size > 16 ?

**Comment by CaoE** [comment]
@ganler  torch.nn.Conv1d(1, 1, 1, padding=1)(torch.ones(16, 1, 1)) that satisfy use_nnpack will call `nnp_convolution_output` in NNPACK.
nnp_convolution_output doesn't support the case of input padding > = kernel_size: 
https://github.com/Maratyszcza/NNPACK/blob/c07e3a0400713d546e0dea2d5466dd22ea389c73/include/nnpack/validation.h#L43

@jbschlosser  Pytorch might be able to add checks like `validate_convolution_arguments` outside `nnp_convolution_output` of NNPACK to avoid such unsupported cases ?

**Comment by ganler** [comment]
@CaoE Maybe we can just follow https://github.com/pytorch/pytorch/pull/49464 to tweak the `use_nnpack` function (add an if-else statement) in https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/Convolution.cpp#L523?

**Comment by CaoE** [comment]
@ganler  could you please try with this PR https://github.com/pytorch/pytorch/pull/92238.
